### PR TITLE
Fixes: Audit missing entries in ClrEtwAll.man #22790

### PR DESCRIPTION
**Pull request by jorive**
Fixes https://github.com/dotnet/coreclr/issues/22790

### CODE REVIEW

**Comment by jorive** [comment]
Sorry, I just realized I trimmed a lot of white spaces when I saved the file (my text editor default). Please let me know if you want me to undo those changes.

**Comment by vancem** [comment]
Manifests are a bit of a pain because they want to be able to localize any strings that you might give to a user (to the local human language), so things like the names of enums are represented as an ID (like                         RundownPublisher.Module.ManifestMapMessage) which are then defined by something (once per language but in practice we never do anything but english), like this
```
                <string id="RuntimePublisher.Module.ManifestMapMessage" value="Manifest" />
```
So you need to add this definition for each of the new enum values you add.

On top of this, we actually have two providers defined in ClrEtwAll.man.  On is the runtime provider and another is what we call the 'rundown' provider.    The rundown provider is mean only for things like JIT events that need to be enumerated before you 'detach' from a session.   Thus often you event is not there, but in this case it is (since we do rundown modules).  

So you actually have an enum defined for each provider (there is no way of sharing the definition), 

In short, just search for one that already exists (ManifestMapMessage) and basically add new ones right beside them at all the places you find them (you should modify 4 separate locations (ref and def for each provider).

**Comment by vancem** [comment]
P.S I don't care about the whitespace change (GitHub now has a obvious way of ignoring whitespace diffs).

**Comment by jorive** [comment]
@vancem modified 4 different locations

**Comment by jorive** [comment]
"OSX10.12 x64 Checked Innerloop Build and Test" failed a second time, but the leg does not seem very stable. Is this a known issue? This is part of the log:
```log
Running tests...

....................................................................Testing for 64 Bit Platforms 
....................Testing for 64 Bit Platforms 
.............................................hello world.foo 52foo 177777foo 177777foo 52foo 52foo  52foo 52 foo 52foo 052foo 052foo 052foo 52foo 52foo 37777777726foo 37777777726.foo 42foo 65535foo 65535foo 42foo 42foo  42foo 42 foo 42foo 042foo 042foo 42foo 42foo 42foo 4294967254foo 4294967254.foo 1234abfoo 1234abfoo 34abfoo 1234abfoo 1234567887654321foo  1234abfoo 1234ab foo 1234abfoo 01234abfoo 01234abfoo 0x1234abfoo 1234abfoo 1234abfoo ffffffd6foo ffffffd6.foo 1234ABfoo 1234ABfoo 34ABfoo 1234ABfoo 1234567887654321foo  1234ABfoo 1234AB foo 1234ABfoo 01234ABfoo 01234ABfoo 0X1234ABfoo 1234ABfoo 1234ABfoo FFFFFFD6foo FFFFFFD6.foo 2.560000e+02foo 2.560000e+02foo 2.560000e+02foo 2.560000e+02foo 2.560000e+02foo   2.560000e+02foo 2.560000e+02  foo 2.6e+02foo 2.56000000e+02foo 002.560000e+02foo 2.560000e+02foo +2.560000e+02foo  2.560000e+02foo -2.560000e+02foo -2.560000e+02.foo 2.560000E+02foo 2.560000E+02foo 2.560000E+02foo 2.560000E+02foo 2.560000E+02foo   2.560000E+02foo 2.560000E+02  foo 2.6E+02foo 2.56000000E+02foo 002.560000E+02foo 2.560000E+02foo +2.560000E+02foo  2.560000E+02foo -2.560000E+02foo -2.560000E+02.foo 2560.001000foo 2560.001000foo 2560.001000foo 2560.001000foo 2560.001000foo  2560.001000foo 2560.001000 foo 2560.0foo 2560.00100000foo 02560.001000foo 2560.001000foo +2560.001000foo  2560.001000foo -2560.001000foo -2560.001000.foo 2560foo 2560foo 2560foo 2560foo 2560foo  2560foo 2560 foo 3e+03foo 2.6e+03foo 2560.001foo 002560foo 2560.00foo +2560foo  2560foo -2560foo -2560.foo 2560foo 2560foo 2560foo 2560foo 2560foo  2560foo 2560 foo 3E+03foo 2.6E+03foo 2560.001foo 002560foo 2560.00foo +2560foo  2560foo -2560foo -2560.babaaaaa4204242042520524204242042420422.0e+002.010e+002.0E+002.010E+002.02.0103e+02256256256.013E+02256256256.01.foo barfoo barfoo barfoo barfoo barfoo barfoo   barfoo bafoo    bafoo bar  foo 00barfoo (null)foo (null)foo (null)foo (null)foo (null)foo (null).foo barfoo barfoo barfoo barfoo barfoo barfoo   barfoo bafoo    bafoo bar  foo 00barfoo (null)foo (null)foo (null)foo (null)foo (null)foo (null).Testing for 64 Bit Platforms 
00000000000000000000000000123456 0000000000123456 00000000001234560000000000123456 00000000001234560X00000000001234560012345600003456001234561234567887654321.foo  barreally-long-string-that-just-keeps-going-on-and-on-and-on....................useless-filler....................................................useless-filler....................................................useless-filler.................................. barfo barfoo  barfoo  barfoo  barfoo  barfoo  barfoo  barfoo  barfoo  barfoo  barfoo  bar.foo bfoo bfoo cfoo bfoo bfoo     bfoo bfoo b    foo 0000bfoo bfoo b.foo bfoo bfoo cfoo bfoo bfoo     bfoo bfoo b    foo 0000bfoo bfoo b.foo 42foo 65535foo -1foo 42foo 42foo  42foo 42 foo 42foo 042foo 042foo 42foo +42foo  42foo -42foo -42.foo 42foo 65535foo -1foo 42foo 42foo  42foo 42 foo 42foo 042foo 042foo 42foo +42foo  42foo -42foo -42........................Testing for 64 Bit Platforms 
.....................................................Testing for 64 Bit Platforms 
............................................Testing for 64 Bit Platforms 
......foo 52foo 177777foo 177777foo 52foo 52foo  52foo 52 foo 52foo 052foo 052foo 052foo 52foo 52foo 37777777726foo 37777777726.foo 42foo 65535foo 65535foo 42foo 42foo  42foo 42 foo 42foo 042foo 042foo 42foo 42foo 42foo 4294967254foo 4294967254.foo 1234abfoo 1234abfoo 34abfoo 1234abfoo 1234567887654321foo  1234abfoo 1234ab foo 1234abfoo 01234abfoo 01234abfoo 0x1234abfoo 1234abfoo 1234abfoo ffffffd6foo ffffffd6.foo 1234ABfoo 1234ABfoo 34ABfoo 1234ABfoo 1234567887654321foo  1234ABfoo 1234AB foo 1234ABfoo 01234ABfoo 01234ABfoo 0X1234ABfoo 1234ABfoo 1234ABfoo FFFFFFD6foo FFFFFFD6.foo 2.560000e+02foo 2.560000e+02foo 2.560000e+02foo 2.560000e+02foo 2.560000e+02foo   2.560000e+02foo 2.560000e+02  foo 2.6e+02foo 2.56000000e+02foo 002.560000e+02foo 2.560000e+02foo +2.560000e+02foo  2.560000e+02foo -2.560000e+02foo -2.560000e+02.foo 2.560000E+02foo 2.560000E+02foo 2.560000E+02foo 2.560000E+02foo 2.560000E+02foo   2.560000E+02foo 2.560000E+02  foo 2.6E+02foo 2.56000000E+02foo 002.560000E+02foo 2.560000E+02foo +2.560000E+02foo  2.560000E+02foo -2.560000E+02foo -2.560000E+02.foo 2560.001000foo 2560.001000foo 2560.001000foo 2560.001000foo 2560.001000foo  2560.001000foo 2560.001000 foo 2560.0foo 2560.00100000foo 02560.001000foo 2560.001000foo +2560.001000foo  2560.001000foo -2560.001000foo -2560.001000.foo 2560foo 2560foo 2560foo 2560foo 2560foo  2560foo 2560 foo 3e+03foo 2.6e+03foo 2560.001foo 002560foo 2560.00foo +2560foo  2560foo -2560foo -2560.foo 2560foo 2560foo 2560foo 2560foo 2560foo  2560foo 2560 foo 3E+03foo 2.6E+03foo 2560.001foo 002560foo 2560.00foo +2560foo  2560foo -2560foo -2560.babaaaaa4204242042520524204242042420422.0e+002.010e+002.0E+002.010E+002.02.0103e+02256256256.013E+02256256256.01.foo barfoo barfoo barfoo barfoo barfoo barfoo   barfoo bafoo    bafoo bar  foo 00barfoo (null)foo (null)foo (null)foo (null)foo (null)foo (null).foo barfoo barfoo barfoo barfoo barfoo barfoo   barfoo bafoo    bafoo bar  foo 00barfoo (null)foo (null)foo (null)foo (null)foo (null)foo (null).Testing for 64 Bit Platforms 
00000000000000000000000000123456 0000000000123456 00000000001234560000000000123456 00000000001234560X00000000001234560012345600003456001234561234567887654321.foo  barreally-long-string-that-just-keeps-going-on-and-on-and-on....................useless-filler....................................................useless-filler....................................................useless-filler.................................. barfo barfoo  barfoo  barfoo  barfoo  barfoo  barfoo  barfoo  barfoo  barfoo  barfoo  bar.foo bfoo bfoo cfoo bfoo bfoo     bfoo bfoo b    foo 0000bfoo bfoo b.foo bfoo bfoo cfoo bfoo bfoo     bfoo bfoo b    foo 0000bfoo bfoo b.foo 42foo 65535foo -1foo 42foo 42foo  42foo 42 foo 42foo 042foo 042foo 42foo +42foo  42foo -42foo -42.foo 42foo 65535foo -1foo 42foo 42foo  42foo 42 foo 42foo 042foo 042foo 42foo +42foo  42foo -42foo -42..............Testing for 64 Bit Platforms 
.................Testing for 64 Bit Platforms 
..............................hello world.................................Testing for 64 Bit Platforms 
.................Testing for 64 Bit Platforms 
....................Testing for 64 Bit Platforms 
.................Testing for 64 Bit Platforms 
....................................................................................................................GetFileSize: ERROR -> Call to SetEndOfFile failed with 112.


FAILED: file_io/GetFileSize/test1/paltest_getfilesize_test1. Exit code: 1

.GetFileSizeEx: ERROR -> SetEndOfFile failed due to lack of disk space


FAILED: file_io/GetFileSizeEx/test1/paltest_getfilesizeex_test1. Exit code: 1

............................................SetFilePointer: ERROR -> SetEndOfFile failed due to lack of disk space

FAILED: file_io/SetFilePointer/test5/paltest_setfilepointer_test5. Exit code: 1

.SetFilePointer: ERROR -> Call to SetEndOfFile failed with error code: 112

FAILED: file_io/SetFilePointer/test6/paltest_setfilepointer_test6. Exit code: 1

.SetFilePointer: ERROR -> Call to SetEndOfFile failed with error code: 112

FAILED: file_io/SetFilePointer/test7/paltest_setfilepointer_test7. Exit code: 1

..................................Testing for 64 Bit Platforms 
....................GlobalMemoryStatusEx:
    ullTotalPhys: 17179869184
    ullAvailPhys: 7705341952
    ullTotalVirtual: 140737488355328
    ullAvailVirtual: 7705341952
    ullTotalPageFile: 1073741824
    ullAvailPageFile: 1028915200
    ullAvailExtendedVirtual: 0
    dwMemoryLoad: 54
...Global Counter Value at the end of the test 0 
...............................
Test #1...

We are testing under Non-Win32 environment.

This test has passed.
.......string comparison passed.
..string comparison passed.
..Producer produces A.
Producer produces B.
Producer produces C.
Producer produces D.
Producer produces E.
Producer produces F.
Producer produces G.
Producer produces H.
Producer produces I.
	Consumer consumes A.
Producer produces J.
	Consumer consumes B.
Producer produces K.
	Consumer consumes C.
Producer produces L.
	Consumer consumes D.
Producer produces M.
	Consumer consumes E.
Producer produces N.
	Consumer consumes F.
Producer produces O.
	Consumer consumes G.
Producer produces P.
	Consumer consumes H.
Producer produces Q.
	Consumer consumes I.
Producer produces R.
	Consumer consumes J.
Producer produces S.
	Consumer consumes K.
Producer produces T.
	Consumer consumes L.
Producer produces U.
	Consumer consumes M.
Producer produces V.
	Consumer consumes N.
Producer produces W.
	Consumer consumes O.
Producer produces X.
	Consumer consumes P.
Producer produces Y.
	Consumer consumes Q.
Producer produces Z.
	Consumer consumes R.
	Consumer consumes S.
	Consumer consumes T.
	Consumer consumes U.
	Consumer consumes V.
	Consumer consumes W.
	Consumer consumes X.
	Consumer consumes Y.
	Consumer consumes Z.
producerItems and consumerItems arrays match.  All 26
items were produced and consumed in order.
Test passed.
.Producer produces A.
	Consumer consumes A.
Producer produces B.
	Consumer consumes B.
Producer produces C.
	Consumer consumes C.
Producer produces D.
	Consumer consumes D.
Producer produces E.
	Consumer consumes E.
Producer produces F.
	Consumer consumes F.
Producer produces G.
	Consumer consumes G.
Producer produces H.
	Consumer consumes H.
Producer produces I.
	Consumer consumes I.
Producer produces J.
	Consumer consumes J.
Producer produces K.
	Consumer consumes K.
Producer produces L.
	Consumer consumes L.
Producer produces M.
	Consumer consumes M.
Producer produces N.
	Consumer consumes N.
Producer produces O.
	Consumer consumes O.
Producer produces P.
	Consumer consumes P.
Producer produces Q.
	Consumer consumes Q.
Producer produces R.
	Consumer consumes R.
Producer produces S.
	Consumer consumes S.
Producer produces T.
	Consumer consumes T.
Producer produces U.
	Consumer consumes U.
Producer produces V.
	Consumer consumes V.
Producer produces W.
	Consumer consumes W.
Producer produces X.
	Consumer consumes X.
Producer produces Y.
	Consumer consumes Y.
Producer produces Z.
	Consumer consumes Z.
producerItems and consumerItems arrays match.  All 26
items were produced and consumed in order.
Test passed.
.Producer produces A.
Producer produces B.
Producer produces C.
Producer produces D.
Producer produces E.
Producer produces F.
Producer produces G.
Producer produces H.
Producer produces I.
	Consumer consumes A.
Producer produces J.
	Consumer consumes B.
Producer produces K.
	Consumer consumes C.
Producer produces L.
	Consumer consumes D.
Producer produces M.
	Consumer consumes E.
Producer produces N.
	Consumer consumes F.
Producer produces O.
	Consumer consumes G.
Producer produces P.
	Consumer consumes H.
Producer produces Q.
	Consumer consumes I.
Producer produces R.
	Consumer consumes J.
Producer produces S.
	Consumer consumes K.
Producer produces T.
	Consumer consumes L.
Producer produces U.
	Consumer consumes M.
Producer produces V.
	Consumer consumes N.
Producer produces W.
	Consumer consumes O.
Producer produces X.
	Consumer consumes P.
Producer produces Y.
	Consumer consumes Q.
Producer produces Z.
	Consumer consumes R.
	Consumer consumes S.
	Consumer consumes T.
	Consumer consumes U.
	Consumer consumes V.
	Consumer consumes W.
	Consumer consumes X.
	Consumer consumes Y.
	Consumer consumes Z.
producerItems and consumerItems arrays match.  All 26
items were produced and consumed in order.
Test passed.
.Producer produces A.
	Consumer consumes A.
Producer produces B.
	Consumer consumes B.
Producer produces C.
	Consumer consumes C.
Producer produces D.
	Consumer consumes D.
Producer produces E.
	Consumer consumes E.
Producer produces F.
	Consumer consumes F.
Producer produces G.
	Consumer consumes G.
Producer produces H.
	Consumer consumes H.
Producer produces I.
	Consumer consumes I.
Producer produces J.
	Consumer consumes J.
Producer produces K.
	Consumer consumes K.
Producer produces L.
	Consumer consumes L.
Producer produces M.
	Consumer consumes M.
Producer produces N.
	Consumer consumes N.
Producer produces O.
	Consumer consumes O.
Producer produces P.
	Consumer consumes P.
Producer produces Q.
	Consumer consumes Q.
Producer produces R.
	Consumer consumes R.
Producer produces S.
	Consumer consumes S.
Producer produces T.
	Consumer consumes T.
Producer produces U.
	Consumer consumes U.
Producer produces V.
	Consumer consumes V.
Producer produces W.
	Consumer consumes W.
Producer produces X.
	Consumer consumes X.
Producer produces Y.
	Consumer consumes Y.
Producer produces Z.
	Consumer consumes Z.
producerItems and consumerItems arrays match.  All 26
items were produced and consumed in order.
Test passed.
.Test Passed
..Producer produces A.
Producer produces B.
Producer produces C.
Producer produces D.
Producer produces E.
Producer produces F.
Producer produces G.
Producer produces H.
Producer produces I.
	Consumer consumes A.
Producer produces J.
	Consumer consumes B.
Producer produces K.
	Consumer consumes C.
Producer produces L.
	Consumer consumes D.
Producer produces M.
	Consumer consumes E.
Producer produces N.
	Consumer consumes F.
Producer produces O.
	Consumer consumes G.
Producer produces P.
	Consumer consumes H.
Producer produces Q.
	Consumer consumes I.
Producer produces R.
	Consumer consumes J.
Producer produces S.
	Consumer consumes K.
Producer produces T.
	Consumer consumes L.
Producer produces U.
	Consumer consumes M.
Producer produces V.
	Consumer consumes N.
Producer produces W.
	Consumer consumes O.
Producer produces X.
	Consumer consumes P.
Producer produces Y.
	Consumer consumes Q.
Producer produces Z.
	Consumer consumes R.
	Consumer consumes S.
	Consumer consumes T.
	Consumer consumes U.
	Consumer consumes V.
	Consumer consumes W.
	Consumer consumes X.
	Consumer consumes Y.
	Consumer consumes Z.
producerItems and consumerItems arrays match.  All 26
items were produced and consumed in order.
Test passed.
.....Iterations:	1000
Threads:	10
[tid=2884096] Thread starting
[tid=2884097] Thread starting
[tid=2884098] Thread starting
[tid=2884099] Thread starting
[tid=2884100] Thread starting
[tid=2884101] Thread starting
[tid=2884102] Thread starting
[tid=2884103] Thread starting
[tid=2884104] Thread starting
[tid=2884105] Thread starting
Created 10 client threads
[tid=2884098] Thread done
[tid=2884103] Thread done
[tid=2884096] Thread done
[tid=2884101] Thread done
[tid=2884105] Thread done
[tid=2884104] Thread done
[tid=2884102] Thread done
[tid=2884100] Thread done
[tid=2884099] Thread done
[tid=2884097] Thread done
.......paltest_duplicatehandle_test8 has been disabled on this platform
...../Users/dotnet-bot/j/workspace/dotnet_coreclr/master/x64_checked_osx10.12_innerloop_prtest/bin/obj/OSX.x64.Checked/src/pal/tests/palsuite/threading/GetCurrentProcessId/test1/paltest_getcurrentprocessid_test1 has dwProcessId 29665
Passing test as dwProcessId is > 0
.paltest_getcurrentthread_test1 has been disabled on this platform
.paltest_getcurrentthread_test2 has been disabled on this platform
...........ReleaseMutex() test returned 0.
Test passed.
............................Global Counter Value: 1 
Global Counter Value: 2 
.Global Counter Value: 1G l
obal Counter Value: 2 
.Global Counter Value: Wait for Single Object (hEvent) was successful.
100000 
Wait for thread was successful
..
 Starting functional  eventing APIs tests  

 All eventing APIs were fired succesfully 

Finished running PAL tests.

The following test(s) failed:
file_io/GetFileSize/test1/paltest_getfilesize_test1. Exit code: 1
file_io/GetFileSizeEx/test1/paltest_getfilesizeex_test1. Exit code: 1
file_io/SetFilePointer/test5/paltest_setfilepointer_test5. Exit code: 1
file_io/SetFilePointer/test6/paltest_setfilepointer_test6. Exit code: 1
file_io/SetFilePointer/test7/paltest_setfilepointer_test7. Exit code: 1

PAL Test Results:
  Passed: 768
  Failed: 5
```

**Comment by jorive** [comment]
@janvorli @mikem8361 Would you know if the OSX failure a know issue? I just retried it for a third time (previous two failures were different: Jenkins disconnected, and the log above).

**Comment by mikem8361** [comment]
Not that I know.  Those PAL test failures look "real" i.e. I don't think they are some kind of intermittent network error.

**Comment by mjsabby** [comment]
I really think we should create an event that spits out the manifest when the provider is activated, much like EventSource. @noahfalk and I talked about it. Otherwise every tool has to keep its manifest up to date with this file in the coreclr repo.

The manifest is pretty large, but the EventSource protocol supports chunking the schema, so it would be a matter of doing the work to emit it on provider load.

**Comment by vancem** [comment]
@mjsabby - we are mostly there in the sense that EventPipe makes the native events look like EventSource events (and describes them to the EventPipe).    That of course is independent of this change...

**Comment by jorive** [comment]
@dotnet/dnceng Are the OSX failures known issues? They seem unrelated to my change.

**Comment by MattGal** [comment]
@jorive taking a look...

**Comment by MattGal** [comment]
@jorive I took a look and the evidence is fairly convincing that this change somehow breaks the tests.  Both have run multiple times on the exact machines they ran on when they failed for you, and in both cases the failures don't imply the machines they're running on:

For the CoreMangLib.system test:
```
      /Users/dotnet-bot/dotnetbuild/work/b7b88e15-9fd2-407f-86fc-cfdcff4f6e02/Payload/corerun environment_version.exe ''
      Environment.Version = 4.0.30319.42000
      ERROR: Version is hardcoded .NET Framework version.
      Expected: 100
      Actual: 255
      END EXECUTION - FAILED
```
I even checked out both machines this ran on, neither had special environment variables set, AFAIK this is a corerun-provided value?

For the  JIT.Intrinsics tests, I'm really unsure how this could be caused by machine state, even intentionally.
```
      BinaryPrimitives.ReverseEndianness(sign-extended Int16) failed.
      Input:    0x00009876
      Output:   0xFFFF7698
      Expected: 0x00007698
      Expected: 100
      Actual: 0
      END EXECUTION - FAILED
```

Given the same exact tests are passing in master and for other folks' PRs on the regular, I suggest rerunning then debugging if it reproduces.

**Comment by jorive** [comment]
The test that was previously failing is now passing after rebasing (it was nothing to do with this PR), but now the Jenkins queue is full.

**Comment by jorive** [comment]
@MattGal Is there any hope these pending legs will run, or should I restart them?

**Comment by MattGal** [comment]
@jorive ci.dot.net seems healthy and is doing OK, but those queue depths are pretty nuts.  We've been having some overloading issues with the Helix API, that may be related.    

@mmitche , ideas?

**Comment by mmitche** [comment]
It's probably a holdover from a reboot. @dotnet-bot test this please

**Comment by jorive** [comment]
```log
DownloadError: Unable to download https://helixexternaldrops.blob.core.windows.net/helix-job-38b461f8-80bf-47b0-5375ebb122234c80a7c79f858d683697/50712321-9f18-4e90-99b3-223292624321.zip after retrying. Execution may fail as a result
```

**Comment by jorive** [comment]
```log
DownloadError: Unable to download https://helixexternaldrops.blob.core.windows.net/helix-job-f86ea0dc-1de7-4c11-7829060a4f8d4dac9688bc379a813758/c6c7f7c8-adde-4703-83bb-8bd8a636be34.zip after retrying. Execution may fail as a result. 
```
@MattGal @mmitche I'm getting this error now. Is Helix working? Is the error expected? Can I ignore this error?

**Comment by hoyosjs** [comment]
Failure to download Helix payloads are part of infrastructure issues tracked at dotnet/core-eng#5391

**Comment by jorive** [comment]
I'll be merging these changes. Failures are due to infrastructure issues.
